### PR TITLE
Added 'validateValue' method to 'CBooleanValidator' class, enh #3812

### PR DESCRIPTION
**Pull request by ua2004**
#3812

### CODE REVIEW

File: framework/validators/CBooleanValidator.php [hunk 1]
```diff
@@ -66,6 +66,25 @@ protected function validateAttribute($object,$attribute)
...
+	 * Validates a static value to see if it is a valid boolean.
+	 * This method is provided so that you can call it directly without going
+	 * through the model validation rule mechanism.
+	 * @param mixed $value the value to be validated
+	 * @return boolean whether the value is a valid boolean
+	 * @since 1.1.17
+	 */
+	public function validateValue($value)
+	{
+		if($this->allowEmpty && $this->isEmpty($value))
```
Comment (cebe) [inline framework/validators/CBooleanValidator.php]: this check should not be done here, and the method could be used in validateAttribute then.

see also #3764

Comment (ua2004) [inline framework/validators/CBooleanValidator.php]: @cebe 'validateAttribute' now uses 'validateValue' method. However, why shouldn't this check be done here? The same check is used in 'validateValue' method in CEmailValidator class.

Comment (cebe) [inline framework/validators/CBooleanValidator.php]: it was accidentally introduced in email validator which broke BC and does not make sense.

Comment (ua2004) [inline framework/validators/CBooleanValidator.php]: But then where is 'allowEmpty' parameter considered?

Comment (cebe) [inline framework/validators/CBooleanValidator.php]: only in validateAttribute

**Comment by samdark** [comment]
@cebe is it ready to be merged?

**Comment by cebe** [comment]
not sure, need to check it.

**Comment by cebe** [comment]
Merged, with adjustments, thanks!

**Pull request closed without merging**